src/macaron/artifact/local_artifact.py

```diff
@@ -247,3 +252,53 @@ def get_local_artifact_dirs(
         )
 
     raise LocalArtifactFinderError(f"Unsupported PURL type {purl_type}")
+
+
+def get_local_artifact_hash(purl: PackageURL, artifact_dirs: list[str], hash_algorithm_name: str) -> str | None:
```

Comment:
Is this function intended to only work with maven PackageURL or we are going to support Python PackagePURL in the future ?

Reply:
We plan to support other ecosystems in the future as well. GitHub attestation can be used for any type of package.

Reply:
We plan to support other ecosystems in the future as well. GitHub attestation can be used for any type of package.